claims.go

```diff
@@ -400,33 +315,46 @@ func (p *Claims) validateHwVersion() error {
 	return nil
 }
 
-func (p *Claims) validateNonce() error {
-	if p.Nonce == nil {
-		return fmt.Errorf("missing mandatory nonce")
+func (c *Claims) validateNonce(expectedProfile string) error {
+	var nonce []byte
+	switch expectedProfile {
+	case PSA_PROFILE_1:
+		if c.LegacyNonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		nonce = *c.LegacyNonce
+	case PSA_PROFILE_2:
+		if c.Nonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		if c.Nonce.Len() != 1 {
+			return errors.New("there must be exactly one nonce")
+		}
+		nonce = c.Nonce.GetI(0)
+	default:
+		return fmt.Errorf("unknown profile: %s", expectedProfile)
 	}
 
-	nonce := ([]eat.Nonce)(*p.Nonce)[0]
-
-	if err := isPSAHashType(nonce.Get()); err != nil {
-		return fmt.Errorf("invalid nonce %s", err.Error())
+	if err := isPSAHashType(nonce); err != nil {
+		return fmt.Errorf("invalid nonce: %w", err)
 	}
 
 	return nil
 }
 
-func (p *Claims) validateSwComponents() error {
-	if p.NoSwMeasurements == 1 {
-		if len(p.SwComponents) != 0 {
+func (c *Claims) validateSwComponents() error {
+	if c.NoSwMeasurements == 1 {
+		if len(c.SwComponents) != 0 {
 			return fmt.Errorf("no-software-measurements and software-components are mutually exclusive")
 		}
 		return nil
 	}
 
-	if len(p.SwComponents) == 0 {
+	if len(c.SwComponents) == 0 {
 		return fmt.Errorf("no software-components found")
 	}
 
-	for i, c := range p.SwComponents {
+	for i, c := range c.SwComponents {
```

Comment:
Is this an error, that sw component variable  is accessed using the same name as that of Claims ?? 
Please clarify..., better to use a local var as sw

Reply:
It is not an error, but it is visually confusing.  I will change this.